eng/tools/oav-runner/src/runner.ts

```diff
@@ -149,16 +156,20 @@ export async function processFilesToSpecificationList(
   for (const file of files) {
     const absoluteFilePath = path.join(rootDirectory, file);
 
-    // if the file is an example, we need to find the swagger file that references it
-    if (example(file)) {
+    // if the file is an example, under "preview" or "stable" (but not the TypeSpec source folder),
+    // we need to find the swagger file that references it
+    if (example(file) && (preview(file) || stable(file))) {
       /*
         examples exist in the same directory as the swagger file that references them:
 
         path/to/swagger/2024-01-01/examples/example.json <-- this is an example file path
         path/to/swagger/2024-01-01/swagger.json <-- we need to identify this file if it references the example
         path/to/swagger/2024-01-01/swagger2.json <-- and do nothing with this one
       */
-      const swaggerDir = path.dirname(path.dirname(file));
+      const swaggerDir = path.relative(
```

Comment:
Instead of hardcoding `dirname(dirname())`, we now use `untilLastSegment("examples")`, which should always return the path until (but excluding) the last occurrence of "examples".  Even if the example is nested arbitrarily deep under "examples".

`untilLastSegment()` always returns an absolute path, so it needs to be resolved against `rootDirectory` to get back to a relative path, which is expected by later code.  In a later PR, this code will be updated to use absolute paths as much as possible, per our design guidelines.